HypothesisArchitecture.asciidoc

```diff
@@ -0,0 +1,38 @@
+
+
+# HDD Architecture
+
+Distributed systems have their own set of challenges. Breaking down a big difficult problem into smaller easier problems still faces the non trivial task of making those smaller problems work together as an integrated unit at scale. Notwithstanding all of this, it remains still an overall good strategy to divide and conquer, but it is important to strive for simplicity while avoiding over-simplification. In order to do this it is necessary to define sensible subsystem boundaries that focus on avoiding dependencies and fulfill clear responsibilities. Well chosen and defined subsystem boundaries make it easier to both manage and incrementally deploy due to minimal coupling.
+ 
+For the hypothesis engine, 3 such  primary autonomous subsystem boundaries have been identified. Communication between these subsystems is done through api endpoints that send/receive data via JSON payloads over http(s) transport .
+
+image::https://github.com/mik-dass/fabric8-hypothesis-documentation/blob/walk_doc/Images/architecture.png[]
+
+1 User Interface(s) : 
+The user interface is the primary means to orchestrate experiments, define metrics and visualize or view collected experimental data via an api. One or more user interfaces can be used.
+ 
+2 Supervisor/Scout Master : 
+The supervisor/scout-master is focused on the collection, aggregation and storage of data obtained from scouts/sensors. The context of received information is determined by supervisor managed experiments. The supervisor becomes the heart and brains of the observable platform, exposed as a service through several api endpoints. In summary the responsibilities are as follows:
+
+* Expose an experiment management api. This api is the mechanism used to create and coordinate experiments. This coordination includes configuration, execution and lifecycle semantics of isolated experiments. Another management aspect is the user driven design of schemas that serve as contracts for measurement data transmitted to the supervisor. Experiments are the means to collect measurements provided by scouts deployed and distributed throughout the system.  This data collection activity is responsible for validating, cleansing, transforming, filtering, correlating, aggregating and potentially storing ( for non transient ) collected data at predetermined intervals in an isolated manner. Repo link : https://github.com/fabric8-hdd/fabric8-hypothesis-manager-api
+* Expose a data retrieval api to make available experimental data. Repo link : https://github.com/fabric8-hdd/fabric8-hypothesis-data-api
+* Expose a data collection api that serves as the central destination for data sent by observing scouts. Repo link : https://github.com/fabric8-hdd/fabric8-hypothesis-measurements-api
+
```

Comment:
We need a way to say that due to the differences in functionality and scaling requirements, each of the 3 different api classifications become candidates for autonomous micro-services and accordingly they have been maintained as part of 3 different repos(then repos being different for each other actually gets a justification)..
I would also be happy to see a diagram that shows this division to make things more clearer...

Reply:
As we discussed, I was talking about the diagram that shows the responsibilities of supervisor into 3 + 1 micro-services..

Reply:
Ok But I will ping @mitchharpur for that diagram and add it later